ov/server_hardware.go

```diff
@@ -359,3 +479,231 @@ func (c *OVClient) GetServerFirmwareByUri(uri utils.Nstring) (ServerFirmware, er
+		t   *Task
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	log.Debugf("REST : %s \n %+v\n", uri, operation)
+	data, err := c.RestAPICall(rest.PATCH, uri, operation)
+	if err != nil {
+		log.Errorf("Error while doing Patch: %s", err)
+		return err
+	}
+
+	log.Debugf("Response of Patch %s", data)
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		log.Errorf("Error with task un-marshal: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+// Performs patch operation to update SH attributes
+func (c *OVClient) Patch(id string, operation []PatchData) error {
+	var (
+		uri = "/rest/server-hardware/" + id
+		t   *Task
+	)
+	// refresh login
+	c.RefreshLogin()
+	c.SetAuthHeaderOptions(c.GetAuthHeaderMap())
+	log.Debugf("REST : %s \n %+v\n", uri, operation)
+	data, err := c.RestAPICall(rest.PATCH, uri, operation)
+	if err != nil {
+		log.Errorf("Error while doing Patch: %s", err)
+		return err
+	}
+
+	log.Debugf("Response of Patch %s", data)
+	if err := json.Unmarshal([]byte(data), &t); err != nil {
+		log.Errorf("Error with task un-marshal: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+// Reset iLO
+func (c *OVClient) SetMpState(serverHardwareId string, value string) error {
+	patchOperation := PatchData{
+		Op:    "replace",
+		Path:  "/mpState",
+		Value: value,
+	}
+	operation := []PatchData{patchOperation}
+	log.Debugf("Update server Hardware's mpState\n")
+	err := c.Patch(serverHardwareId, operation)
+	if err != nil {
+		log.Errorf("Error while updating mpState: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+// Update the server into/out of maintenance mode
+func (c *OVClient) SetMaintenanceMode(serverHardwareId string, value string) error {
+	patchOperation := PatchData{
+		Op:    "replace",
+		Path:  "/maintenanceMode",
+		Value: value,
+	}
+	operation := []PatchData{patchOperation}
+	log.Debugf("Update server Hardware's maintenance mode\n")
+	err := c.Patch(serverHardwareId, operation)
+	if err != nil {
+		log.Errorf("Error while updating maintenance mode: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+// Turn the server UID light On/Off
+func (c *OVClient) SetUidState(serverHardwareId string, value string) error {
+	patchOperation := PatchData{
+		Op:    "replace",
+		Path:  "/uidState",
+		Value: value,
+	}
+	operation := []PatchData{patchOperation}
+	log.Debugf("Update server Hardware's uidState\n")
+	err := c.Patch(serverHardwareId, operation)
+	if err != nil {
+		log.Errorf("Error while updating uidState: %s", err)
+		return err
+	}
+
+	return nil
+}
+
+// Update the power state of the server
+func (c *OVClient) SetPowerState(serverHardwareId string, powerState map[string]interface{}) error {
+	patchOperation := PatchPowerData{
+		Op:    "replace",
+		Path:  "/powerState",
+		Value: powerState,
+	}
+	operation := []PatchPowerData{patchOperation}
+	log.Debugf("Update server Hardware's power state\n")
+	err := c.PatchPowerState(serverHardwareId, operation)
```

Comment:
do really need this patchPowerState()?

Reply:
> > @yuvirani See if you can return operation status or json response for success scenario instead of returning as "nil"
> > -- Since we update the object, we dont return the same response again. We can access object attributes to check values updated correctly.
> > And also I can't see any unit tests written for new functionality. Don;t they need to be added.
> > -- I will add them

I think you are saying for PATCH operations, but I am talking about POST operation